src/views/TradingCompetition/hooks/useGetCompetitionState.ts

```diff
@@ -0,0 +1,36 @@
+import { useState, useEffect } from 'react'
+import { useInitialBlock } from '../../../state/hooks'
+import { CompetitionState } from '../config'
+
+const useGetCompetitionState = () => {
+  const [targetBlock, setTargetBlock] = useState(null)
+  const [competitionState, setCompetitionState] = useState(null)
+  const [isLoading, setIsLoading] = useState(true)
+  const currentBlock = useInitialBlock()
+
+  useEffect(() => {
+    const liveBlock = CompetitionState.LIVE.startBlock
+    const finishedBlock = CompetitionState.FINISHED.startBlock
+
+    if (currentBlock < liveBlock) {
```

Comment:
This assumes that the page is in a 'registration' state at any time before the competition starting. While registration does have a block start date (and therefore there is actually a 'preregistration state') - the page won't be live prior to that, so I didn't think it necessary to write logic for that condition.